### PR TITLE
Adding alert rule with a syntax error for testing

### PR DESCRIPTION
**Pull request by Abuelodelanada**
This simple PR adds an alert rule with syntax error for testing purposes

### CODE REVIEW

File: tests/sample_rule_files/free-standing/error/alert.rule [hunk 1]
```diff
@@ -0,0 +1,20 @@
+groups:
```
Comment (sed-i) [inline tests/sample_rule_files/free-standing/error/alert.rule:1]: Is the subdir needed? I.e. would `free-standing/alert_with_syntax_error_in_expr.rule` work?

**Review by sed-i — APPROVED** [review]
I wonder if these kinds of tests would be more manageable if:
- we keep them under the cos bundle repo
- we use cos-config with rule files we write to disk at runtime (git-sync supports local folders for syncing git repos)

File: tests/sample_rule_files/free-standing/error/alert.rule (hunk 1 above)
Comment (Abuelodelanada) [reply tests/sample_rule_files/free-standing/error/alert.rule:1]: I believe it's better to have them in a separate directory so we can use cos-config in itests:

```
        ops_test.model.deploy(
            "ch:cos-configuration-k8s",
            application_name=rules_app3,
            channel="edge",
            config={
                "git_repo": "https://github.com/canonical/loki-k8s-operator",
                "git_branch": "main",
                "loki_alert_rules_path": "tests/sample_rule_files/free-standing/error",
            },
        ),
    )
```

**Comment by Abuelodelanada** [comment]
> I wonder if these kinds of tests would be more manageable if:
> 
>     * we keep them under the cos bundle repo

I do not know... checking alert rules is something Loki lib does...

> 
>     * we use cos-config with rule files we write to disk at runtime (git-sync supports local folders for syncing git repos)

Oh, if this is possible, it is a better idea. Do we have an example of this?

**Comment by mmanciop** [comment]
By which test are these used? It’s unclear from the PR if they are used

**Comment by Abuelodelanada** [comment]
> By which test are these used? It’s unclear from the PR if they are used

By a test I'm writing in which cos-config clones loki repo, so I need the error alert rule in `main` beforehand.

**Comment by mmanciop** [comment]
Mmm this sounds like a weird setup. Why not putting these data into a different, protected branch of cos-config instead?

**Review by dstathis — APPROVED** [review]

**Comment by simskij** [comment]
So am I correct in understanding that this actually is to be used by cos-config, which is then pointed here to get the alert rules?

My initial reaction is that this sounds unnecessarily complicated. Why is cos-config even needed in this case?

**Comment by simskij** [comment]
> Mmm this sounds like a weird setup. Why not putting these data into a different, protected branch of cos-config instead?

I would say it's preferable if the test resources live near the code that uses them. If anything, I would like to avoid involving cos-config at all for these tests, although I'm not sure whether that would be feasible.

**Review by simskij — APPROVED** [review]
lgtm. let's get this merged. we can iterate on the specifics later.

**Pull request merged**